lib/go.coffee

```diff
@@ -42,6 +45,55 @@ class Go
     result = gopathConfig if not environmentOverridesConfig and gopathConfig? and gopathConfig.trim() isnt ''
     result = gopathConfig if result is '' and gopathConfig? and gopathConfig.trim() isnt ''
     result = result.replace('\n', '').replace('\r', '')
+
+    # Whenever we build GOPATH, go up the directory tree to identify where there
+    # might be a Godeps directory. If we find it in any parent/ancestor directory,
+    # add it to GOPATH.
+    goDepDir = null
+    if cwd?
+      goDepDir = @goDepLookupCache.get(cwd)
+      if goDepDir is undefined
+        minLastSeparator = 0
+        if os.platform() is 'win32' and cwd.contains(':\\')
+          minLastSeparator = 2
+
+        goDepCheck = cwd
+        lastSeparator = cwd.length
+        while lastSeparator > minLastSeparator
+          goDepCheck = cwd.substring(0, lastSeparator)
+          if fs.existsSync(goDepCheck + path.sep + "Godeps")
+            goDepDir = goDepCheck + path.sep + "Godeps" + path.sep + "_workspace"
+            break
+          lastSeparator = goDepCheck.lastIndexOf(path.sep)
+        @goDepLookupCache.set(cwd, goDepDir)
+
+    if goDepDir?
+      result = goDepDir + path.delimiter + result
+
+    # Whenever we build GOPATH, go up the directory tree to identify where there
+    # might be a .gopack directory. If we find it in any parent/ancestor directory,
```

Comment:
What tool is using the `.gopack` directory? I couldn't find it. https://github.com/d2fn/gopack seems to use a `gopack.config` file...